### DISCUSSION TITLE
How to support shared sidebars

### DISCUSSION THREAD

**Discussion by glitch-txs** [facebook/docusaurus#10184] [Q&A]
Hi! I have the following setup

```ts
export default {
  tutorialSidebar:['doc1', 'doc2', 'commonDoc'],
  apiSidebar: ['doc3', 'doc4', 'commonDoc'],
};
```

My current problem is that when I'm in the tutorial page (using `tutorialSidebar`) and I press `commonDoc` it will change the first to pages from `doc1` and `doc2` to `doc3` and `doc4`, meaning that the sidebar displayed is the `apiSidebar` now.

How can I keep the sidebar static when going to the `commonDoc` page? either if I'm in the tutorial or api section I want the first two pages to **not** change when I select `commonDoc`.

**Comment by slorber** [comment]
It's not ideal to have a common doc bound to 2 distinct sidebars. We have to pick one to display when you browse that doc and it might not be the one you want.

You can use a sidebar item of type "ref" so that it appears in a sidebar but that sidebar is not displayed when you browse the doc.

https://docusaurus.io/docs/next/sidebar/multiple-sidebars#sidebar-item-ref

You can also use `displayed_sidebar: tutorialSidebar` front matter on the common doc

**Reply by glitch-txs** [reply]
Why it's not ideal? Is there any workaround?

So I managed to build the selector for frameworks, but some documents are shared even though the frameworks are diferent because it's the same SDK.

**Reply by slorber** [reply]
If a doc is in sidebar1 and sidebar2, then when you browse that doc, which sidebar should we display? 

This introduces an ambiguity and we have to pick a sidebar randomly (what we do) or emit an error.

**Reply by glitch-txs** [reply]
> If a doc is in sidebar1 and sidebar2, then when you browse that doc, which sidebar should we display?

The current one. Would that be difficult to apply?

ATM as a workaround I just recreated multiple times the shared doc and imported the content from the same file

**Reply by slorber** [reply]
> The current one. Would that be difficult to apply?

the notion of "current one" does not exist.

You open an URL `/docs/commonDoc`, which one does it show? It must be one or the other.

Docusaurus does not consider the "navigation" history you came from when deciding which sidebar to display. A doc will always display the same sidebar when you navigate to it. If you want 2 identical docs with distinct URLs and disting sidebars, then you have 2 docs, not one. You can mutualize the content of those 2 thanks to `_common.mdx` partial to avoid duplicating the content.

**Reply by glitch-txs** [reply]
> the notion of "current one" does not exist.

So there's no way to keep track of that with internal state?

**Reply by slorber** [reply]
Keep track of "what"?

When you browse `/docs/commonDoc` we have to display something. There's no state, it's a fresh browser load, it has to be predictable.

**Reply by glitch-txs** [reply]
you can use the path or url queries. I'm using a path to keep track of this

**Reply by slorber** [reply]
A static document has a single url/pathname for which we have to display a given sidebar

To avoid bad loading ux, the sidebar has to be in the static file, and cdns do not usually support serving different static files for query string parameters. If you want to display different sidebars depending on search parameters, then it's a dynamic feature that we don't provide and that you'll have to implement yourself. The ux would not be ideal IMHO.

---

In general please be concrete. Tell me which url exactly should show which sidebar on a given repro. A code example is more relevant than trying to describe things as text.

**Reply by glitch-txs** [reply]
I have already solved it now copy/pasting the same files multiple times. You can play with it [here](https://walletconnect-docs-git-appkit-walletkit-walletconnect1.vercel.app/appkit/react/core/installation). The Cloud section is the static one, and the framework changes are reflected in the url path.

I guess it's way too complex to implement to offer a customizable solution. Thanks for the answer 👍

**Comment by webprofusion-chrisc** [comment]
For anyone looking for the same sort of thing one trick is to add an additional sidebar config which appears last in the last of sidebar configs, with a title such as "Common" (or anything to indicate that it's shared). Then this title will be shown for any shared documents, rather than showing the wrong product category etc. Example: https://docs.certifytheweb.com/

I guess the alternative would have been to generated new routes per sidebar and duplicate each common document under each one, as duplicating an entire document is obviously not the right approach from a maintenance perspective.

**Reply by slorber** [reply]
Why not use refs as I suggest instead?

Instead of duplicating content, you can also duplicate the docs files but import a shared `_partial.mdx` file in each of those: you would only need to maintain the content once. The difference with refs is that each "clone" will have a distinct URL and will display a distinct sidebar, despite presenting the same content.
